internal/server/proto/server.proto

```diff
@@ -3178,6 +3199,113 @@ message ReleaseExpanded {
   StatusReport latest_status_report = 2;
 }
 
+/********************************************************************
+* Trigger URLs
+********************************************************************/
+
+// The Trigger message is a representation of a URL that can be requested
+// for invoking specific lifecycle operations on a projects applications. These
+// trigger URLs are intended to be used in a CI system for easy configuration.
+// The user is expected to configure and generate the URL ahead of time, and
+// Waypoint will generate a trigger based on the configuration here and return
+// a URL to make requests with.
+message Trigger {
+  reserved 57 to 99; // future operation range
+
+  // uuid generated by Waypoint on creation. Used as the identifier in the URL.
+  string id = 1;
+
+  // name can be user defined, or auto generated.
+  string name = 2;
+
+  // description is user defined, describes the purpose
+  // of the trigger.
+  string description = 3;
+
+  // labels that allow users to group triggers together that are
+  // related. There are no default labels.
+  repeated string labels = 4;
```

Comment:
Curious why this isn't a map<string,string> like other labels.

Reply:
Those labels are more key=>val (literally a map), where as the intention here is a flat list, similar to how you'd label something in GitHub. Just a way to categorize something (like, list me all the trigger URLs for `"test"` or `"production"`, etc).

Reply:
I wonder if we should call these "tags" or something since "labels" is used so heavily throughout WP otherwise to mean a map of key/value.

Reply:
That change makes sense to me!